src/Stream/MemoryStream.tpp

```diff
@@ -325,6 +325,43 @@ namespace Stream
     {
     }
 
+    //
+    // BaseOStream Helper class to enable partial specialization
+    //
+
+    /**
+     * Helper for operator<<(BaseOStream&, const ArgT&), generalized version
+     */
+    template<typename Elem, typename ArgT, typename Enable = void>
+    struct BaseOStreamImpl 
+    {
+      BaseOStream<Elem>& operator()(BaseOStream<Elem>& ostr, const ArgT& arg)
+        /*throw eh::Exception*/
+      {
+        std::ostringstream ss;
+        ss << arg;
+        ostr.append(ss.str().c_str());
+        return ostr;
+      }
+    };
+
+    /**
+     * Helper for operator<<(BaseOStream&, const ArgT&)
+     * specialization for to_chars applicable types
+     */
+    template <typename Elem, typename ArgT>
+    struct BaseOStreamImpl<Elem, ArgT, 
+      decltype(std::to_chars(std::string().data(), std::string().data(), ArgT()), void())> 
+    {
+      BaseOStream<Elem>& operator()(BaseOStream<Elem>& ostr, const ArgT& arg)
+        /*throw eh::Exception*/
+      {
+        // TODO
+        ostr << std::to_string(arg);
```

Comment:
Switch to to_chars

Reply:
switched, but implementation needs to discussed.
to_chars does nothing if [first, last) is shorter than length of converted argument : https://en.cppreference.com/w/cpp/utility/to_chars ("Return value" part).

Reply:
I first calculate length of to_chars.
then apply to_chars.